Duplicati/Library/Backend/Telegram/TelegramBackend.cs

```diff
@@ -0,0 +1,516 @@
+                            EnsureConnected();
+                            var file = m_telegramClient.GetFile(fileLocation, limit, currentOffset).GetAwaiter().GetResult();
+                            stream.Write(file.Bytes, 0, file.Bytes.Length);
+                            currentOffset += file.Bytes.Length;
+                        }
+                        catch (InvalidOperationException e)
+                        {
+                            if (e.Message.Contains("Couldn't read the packet length") == false)
+                            {
+                                throw;
+                            }
+                        }
+                    }
+                },
+                $"{nameof(Get)}");
+        }
+
+        public Task PutAsync(string remotename, string filename, CancellationToken cancelToken)
+        {
+            using (var fs = File.OpenRead(filename))
+            {
+                PutAsync(remotename, fs, cancelToken).GetAwaiter().GetResult();
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public void Get(string remotename, string filename)
+        {
+            using (var fs = new FileStream(filename, FileMode.Create, FileAccess.Write))
+            {
+                Get(remotename, fs);
+            }
+        }
+
+        public void Delete(string remotename)
+        {
+            SafeExecute(() =>
+                {
+                    var channel = GetChannel();
+                    var fileInfo = ListChannelFileInfos().FirstOrDefault(fi => fi.Name == remotename);
+                    if (fileInfo == null)
+                    {
+                        return;
+                    }
+
+                    var request = new TLRequestDeleteMessages
+                    {
+                        Channel = new TLInputChannel
+                        {
+                            ChannelId = channel.Id,
+                            AccessHash = channel.AccessHash.Value
+                        },
+                        Id = new TLVector<int> {fileInfo.MessageId}
+                    };
+
+                    EnsureConnected();
+                    m_telegramClient.SendRequestAsync<TLAffectedMessages>(request).GetAwaiter().GetResult();
+                },
+                $"{nameof(Delete)}({remotename})");
+        }
+
+        public List<ChannelFileInfo> ListChannelFileInfos()
+        {
+            var channel = GetChannel();
+
+            var inputPeerChannel = new TLInputPeerChannel {ChannelId = channel.Id, AccessHash = channel.AccessHash.Value};
+            var result = new List<ChannelFileInfo>();
+            var oldMinDate = 0;
+            var newMinDate = (int?)null;
+
+            while (oldMinDate != newMinDate)
+            {
+                oldMinDate = newMinDate ?? 0;
+                RetrieveMessages(inputPeerChannel, result, oldMinDate);
+                newMinDate = result.Min(cfi => (int)cfi.Date.GetEpochSeconds());
```

Comment:
The `Utility.NormalizeDateTimeToEpochSeconds` method already exists.  Does that work here?  Note that it computes the value in terms of the UTC time.

Reply:
Sorry, I am unable to find the method. Are you sure its' name is correct?

Reply:
Found it, I'll use this, will be ready in few minutes